calc.py

```diff
@@ -112,7 +112,9 @@ def p_program(p):
 
 def p_lines_single_one(p):
     "lines : instruction"
-    p[0] = p[1]
+    node = Instructions(p.lineno(1))
+    node.instructions.insert(0, p[1])
+    p[0] = node
```

Comment:
I think this leads to nested Instructions node everytime there is more than one instruction. If this is just for handling single instruction case then maybe we can handle it in AST constructor?  If root node is not Instructions node then we create Instructions and insert passed root node to it

Reply:
nvmnd